src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4060,6 +4060,45 @@ void C2_MacroAssembler::masked_op(int ideal_opc, int mask_len, KRegister dst,
 }
 
 #ifdef _LP64
+void C2_MacroAssembler::vector_long_to_maskvec(XMMRegister dst, Register src, Register rtmp1,
+                                               Register rtmp2, XMMRegister xtmp, int mask_len,
+                                               int vec_enc) {
+  int index = 0;
+  int vindex = 0;
+  movq(rtmp2, src);
+  mov64(rtmp1, 0x0101010101010101L);
+  pdep(rtmp1, rtmp2, rtmp1);
```

Comment:
For masklen < 8, we could directly generate pdep(rtmp1, src, rtmp1);
rtmp2 is not required in that case.

Reply:
DONE